src/silly_sprites/event_system.hpp

```diff
@@ -0,0 +1,67 @@
+#include "events.hpp"
+#include "types.hpp"
+#include <vector>
+
+namespace sly::event {
+
+    struct EventID final {
+        friend class EventSystem;
+
+    private:
+        usize m_id;
+        EventID(usize id) : m_id{ id } {};
+
+    public:
+        EventID() : m_id{ 0 } {};
+        [[nodiscard]] auto operator<=>(EventID const&) const = default;
+    };
+
+    class EventSystem final {
+    private:
+        std::vector<std::pair<EventID, EventCallbacks>> m_handlers;
```

Comment:
Ah, by the way: We already talked about this and agreed on that `std::map` (or `std::unordered_map`) cannot be used here and that still holds true. But maybe we should look into `std::multimap` or `std::unordered_multimap`. I don't think that it is necessary, but maybe it's nice to have worked with those things :)